binding.go

```diff
@@ -112,6 +112,25 @@ func Bind(obj interface{}, ifacePtr ...interface{}) macaron.Handler {
 	}
 }
 
+// URL works like Bind, but bind fields from args in URL to struct.
+func URL(obj interface{}, ifacePtr ...interface{}) macaron.Handler {
+	return func(ctx *macaron.Context) {
+		var errors Errors
+
+		ensureNotPointer(obj)
+		obj := reflect.New(reflect.TypeOf(obj))
+
+		val := obj.Elem()
+		for k, v := range ctx.AllParams() {
+			field := val.FieldByName(k[1:])
+			if field.IsValid() {
+				field.SetString(v)
```

Comment:
I think assuming the struct field type is always `string` is dangerous (i.e. could lead to panic?) Instead, I believe we should call `setWithProperType` for each URL parameter we found, see examples: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/go-macaron/binding%24%40master++%3D+setWithProperType+-file:_test%5C.go%24+&patternType=literal